Source/AkiSupport/Singleplayer/Patches/Progression/ExperienceGainPatch.cs

```diff
@@ -33,9 +33,16 @@ private static bool IsTargetMethod(MethodInfo mi)
         }
 
         [PatchPrefix]
-        private static void PatchPrefix(ref Profile profile, ref bool isOnline)
+        private static void PatchPrefix(ref Profile profile, ref bool isOnline, ref ExitStatus exitStatus)
         {
-            Logger.LogInfo("PatchPrefix");
+            Logger.LogInfo("PatchPrefix");            
+            // If extracted alive subtract 300 from session experience to account for extract bonus, if below 200 it's a run through
+            if (exitStatus == ExitStatus.Survived)
+            {
+                var xpGained = profile.EftStats.TotalSessionExperience;
+                if (xpGained > 300) xpGained -= 300;
+                if (xpGained < 200) exitStatus = ExitStatus.Runner;
```

Comment:
You shouldn't setup a hard-coded number because all of them is available and free to modify at Aki's "globals.json", also you are missing the 7 minates check.
By reading these values correctly from the Aki server, you should run the following code below:
```Python
var matchEnd = Singleton<Config4>.Instance.Experience.MatchEnd;
int survivedExpRequirement = matchEnd.SurvivedExpRequirement;
int survivedTimeRequirement = matchEnd.SurvivedTimeRequirement;
int survivedReward = matchEnd.SurvivedReward;
```

Reply:
Implemented this but not the SurvivedTimeRequirement. It's 420 seconds at which point players get a 300 xp bonus that puts them over the SurvivedExpRequirement (200) so it would be redundant. I also don't know a good way to get past raid time information after the fact.